### ISSUE TITLE
Incorrect path to download libxml2

### ISSUE THREAD

**Issue by Durlyn** [xbmc/inputstream.ffmpegdirect#182]
Hi!
compiling inputstream.ffmpegdirect from git.
at the stage of searching for the libxml2 library, an error pops up:

CMake Error at libxml2-stamp/download-libxml2.cmake:159 (message)
 Each download failed!

along the way http://xmlsoft.org/sources/libxml2-2.9.10.tar.gz there is no more file with libxml2. how to compile from git now?

**Comment by Durlyn** [comment]
It looks like the libxml2 file is now here https://download.gnome.org/sources/libxml2/2.9/libxml2-2.9.14.tar.xz
corrected in /depends/common/libxml2/libxml2.txt new path and in /depends/common/libxml2/libxml2.sha256 to new sha256 summ, it turned out to compile

**Comment by phunkyfish** [comment]
Thanks for finding this. I’ll update the repo short with this.

**Comment by phunkyfish** [comment]
Hmmm, I just tested this and it’s working. Maybe their server was down temporarily

**Comment by Durlyn** [comment]
the server is running, but https://xmlsoft.org/sources/, then a list of files opens. but there is no libxml2, only libvirt

**Comment by phunkyfish** [comment]
Weird, cause if I click the link it works and a download starts.

**Comment by Durlyn** [comment]
really strange.
wget http://xmlsoft.org/sources/libxml2-2.9.10.tar.gz 
connection to xmlsoft.org (xmlsoft.org)|91.121.203.120|:80... connection established.
HTTP-request sent. The reply is in process.. 403 Forbidden
ERROR 403: Forbidden.

**Comment by phunkyfish** [comment]
Ok, I've PR'd a change to use the kodi mirror instead:

Matrix - https://github.com/xbmc/inputstream.ffmpegdirect/pull/184
Nexus - https://github.com/xbmc/inputstream.ffmpegdirect/pull/183

**Comment by Whisprin** [comment]
I'm trying to build version 19.0.2-Matrix
However, I'm receiving this error related to libxml2-2.9.10

```
libxml2-2.9.10.tar.gz
  does not match expected value
    expected: 'aafee193ffb8fe0c82d4afef6ef91972cbaf5feea100edc2f262750611b4be1f'
      actual: 'f07dab13bf42d2b8db80620cce7419b3b87827cc937c8bb20fe13b8571ee9501'
-- Hash mismatch, removing...
-- Using src='http://mirrors.kodi.tv/build-deps/sources/libxml2-2.9.10.tar.gz'
```

And indeed even after retrying the downloads, hash values from the two mirrors differ:
```
f07dab13bf42d2b8db80620cce7419b3b87827cc937c8bb20fe13b8571ee9501  libxml2-2.9.10-mirrors.kodi.tv.tar.gz
aafee193ffb8fe0c82d4afef6ef91972cbaf5feea100edc2f262750611b4be1f  libxml2-2.9.10-xmlsoft.org.tar.gz
```

The current version expects the hash value from xmlsoft.org but takes the file from mirrors.kodi.tv.

**Comment by phunkyfish** [comment]
Very strange. Because the PR passed CI. Very concerning that the two files have different hashes and have the same version number.

**Comment by Whisprin** [comment]
> Very strange. Because the PR passed CI. Very concerning that the two files have different hashes and have the same version number.

Indeed. Can you replicate the issue? The contents of the two archives differ for me as well (by more than just a Readme or a timestamp)
xmlsoft contains the folder: ./libxml2-2.9.10
kodi contains the folder: ./libxml2-**v**2.9.10
'+ some files existing in just either one of the directories. Has Kodi changed the archive in their mirror after CI passed? 

Diff: https://gist.github.com/Whisprin/8aa04d54570a0c724d7e9f22f1542605

**Comment by phunkyfish** [comment]
That’s a possibility. Anyway with can find this out @wsnipex ?

**Comment by wsnipex** [comment]
no, we don't change any archive contents. I'd rather assume we got "our" tarball directly from https://github.com/xbmc/libxml2

**Comment by phunkyfish** [comment]
Hmmm, so as there are issues with the xmlsoft tarball download how should we store this? Can we crate a subdirectory and store a copy there? Or some other option?

**Comment by Whisprin** [comment]
> no, we don't change any archive contents. I'd rather assume we got "our" tarball directly from https://github.com/xbmc/libxml2

This repo seems to be stuck at version 2.9.4 and doesn't contain the files present in the Kodi archive, e.g. file "CONTRIBUTING". Is there another source for the tarball?

> Hmmm, so as there are issues with the xmlsoft tarball download how should we store this? Can we crate a subdirectory and store a copy there? Or some other option?

It seems that [other](http://rpmfind.net/linux/RPM/fedora/34/x86_64/l/libxml2-2.9.10-10.fc34.x86_64.html) projects are pointing to the xmlsoft mirror as well and that the downtime of xmlsoft was just a temporary hickup. 
Another source is the [GNOME Gitlab](https://gitlab.gnome.org/GNOME/libxml2/-/releases#v2.9.10).

**Comment by phunkyfish** [comment]
I’ll just revert the commit that changed the URL.

**Comment by AlwinEsch** [comment]
On this request https://github.com/xbmc/inputstream.ffmpegdirect/pull/241 becomes it changed to use https://github.com/GNOME/libxml2/archive/refs/tags/v2.11.4.tar.gz, think with GitHub should it work better.
